### ISSUE TITLE
$this->publicKey is immediately overridden

### ISSUE THREAD

**Issue by applibs** [phpseclib/phpseclib#1588]
```
$this->publicKey = null;
$this->publicKey = $this->getPublicKey();
```

**Comment by terrafrost** [comment]
You couldn't have possibly submitted a more useless bug report. I shouldn't have to do ` grep -r "\$this->publicKey = \$this->getPublicKey" .` to find out wtf you're talking about.

Anyway, `$this->publicKey = $this->getPublicKey();` can throw exceptions. So if you already had a CSR loaded and then wanted to load another CSR with a malformed public key, if it wasn't for that initial `$this->publicKey = null` line then `$this->publicKey` would retain the original public key if an exception was thrown.

It's because of shit like this that I made public key objects immutable. Loading correctly formatted keys and then reloading a malformed key with the same object you used to load the first key can result in strange behaviors.

And sure, there are other things I could have done. I could have done this, for example:

```php
try {
    $this->publicKey = $this->getPublicKey();
} catch (\Exception $e) {
    $this->publicKey = null;
    throw $e;
}
```
Basically does the same thing. Is it any better? idk.
